tests/network/port_forward.go

```diff
@@ -0,0 +1,139 @@
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libvmi"
+)
+
+var _ = SIGDescribe("[Serial] Port-forward", func() {
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+
+		tests.BeforeTestCleanup()
+	})
+
+	Context("VMI With masquerade binding", func() {
+		const localPort = 1500
+		var (
+			portForwardCmd *exec.Cmd
+			vmi            *v1.VirtualMachineInstance
+			VMIport        int
+			ports          []v1.Port
+		)
+
+		JustBeforeEach(func() {
+			vmi = createCirrosVMIWithPortsAndBlockUntilReady(virtClient, ports)
+			tests.StartHTTPServer(vmi, VMIport)
+
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+			Expect(vmiPod).ToNot(BeNil())
+			portForwardCmd, err = portForwardCommand(vmiPod, localPort, VMIport)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(portForwardCmd.Start()).To(Succeed())
+		})
+
+		When("performing port-forward from a local port to a VMI's pod declared port", func() {
+			const declaredPort = 1501
+			BeforeEach(func() {
+				ports = append(ports, v1.Port{Port: declaredPort})
+				VMIport = declaredPort
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
+					Expect(killPortForwardCommand(portForwardCmd)).To(Succeed())
+				}()
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", declaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 40*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+
+		When("performing port-forward from a local port to a VMI's pod non-declared port", func() {
+			const nonDeclaredPort = 1502
+			BeforeEach(func() {
+				ports = []v1.Port{}
+				VMIport = nonDeclaredPort
+			})
+
+			It("should reach the vmi", func() {
+				defer func() {
+					Expect(killPortForwardCommand(portForwardCmd)).To(Succeed())
+				}()
+
+				By(fmt.Sprintf("checking that service running on port %d can be reached", nonDeclaredPort))
+				Eventually(func() error {
+					err = exec.Command("curl", fmt.Sprintf("127.0.0.1:%d", localPort)).Run()
+					return err
+				}, 40*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+			})
+		})
+	})
+})
+
+func portForwardCommand(pod *k8sv1.Pod, sourcePort, targetPort int) (*exec.Cmd, error) {
+	_, cmd, err := tests.CreateCommandWithNS(pod.Namespace, tests.GetK8sCmdClient(), "port-forward", pod.Name, fmt.Sprintf("%d:%d", sourcePort, targetPort))
+
+	return cmd, err
+}
+
+func killPortForwardCommand(portForwardCmd *exec.Cmd) error {
+	if portForwardCmd == nil {
+		return nil
+	}
+
+	return portForwardCmd.Process.Kill()
```

Comment:
I see that [here](https://github.com/kubevirt/kubevirt/blob/fd39d9e22a57c6391e03fd8c496e98560d23490f/tests/storage/imageupload.go#L179) there is also a `wait` for the command to be terminated. What is the reason for not having it here as well?
According to the [documentation](https://golang.org/pkg/os/#Process.Kill) - "kill does not wait until the Process has actually exited". So if we want to be sure the port-forwarding is kill before the next command is invoked, I believe we should add a `portForwardCmd.Wait`.

Reply:
To be on the safe side modified the test to use different local ports. That's the only dangerous thing I could think of, but still I don't think we need to wait.
Is that acceptable?

Reply:
Why not using `wait`? IMO, maintaining there is no overlap between tests can be difficult/dangerous. Using `wait` is much simpler.

Reply:
I thought it will be faster, but I guees it is not that significant. Will change to `wait()`

Reply:
[Done](https://github.com/kubevirt/kubevirt/compare/d65db08009e6f838e208f0312576f23305371196..3e5d913addf7506b3c2738401bb7a99677ba1035)